components/organisms/ContributorProfileTab/contributor-profile-tab.tsx

```diff
@@ -94,32 +94,16 @@ const ContributorProfileTab = ({
   const { data: emojis } = useFetchAllEmojis();
 
   const router = useRouter();
-  const { tab } = router.query;
+  const { tab = "highlights" } = router.query as { tab: TabKey };
 
   const hasHighlights = highlights ? highlights.length > 0 : false;
   const [inputVisible, setInputVisible] = useState(false);
 
-  const currentTab = (
-    (Object.keys(tabs).includes(tab as string) && tab) || hasHighlights ? "highlights" : "contributions"
-  ) as TabKey;
   function onTabChange(value: string) {
     const tabValue = value as TabKey;
     setQueryParams({ tab: tabValue } satisfies QueryParams);
   }
 
-  useEffect(() => {
```

Comment:
The use effect kept adding the query param to the browser history. This is not necessary. We can set a default tab even if it's not in the URL.